lib/src/main/kotlin/com/swmansion/starknet/crypto/NativeLoader.kt

```diff
@@ -62,8 +63,28 @@ internal object NativeLoader {
     private fun getLibPath(system: SystemType, architecture: String, name: String): String {
         return when (system) {
             SystemType.MacOS -> "/darwin/$name.dylib"
-            SystemType.Linux -> "/linux/$architecture/$name.so"
-            else -> throw UnsupportedPlatform(operatingSystem.name, architecture)
+            SystemType.Linux -> {
+                // Try to load from AAR
+                val androidAbi = runCatching {
+                    val buildClass = Class.forName("android.os.Build")
+
+                    @Suppress("UNCHECKED_CAST")
+                    val supportedAbis = buildClass.getField("SUPPORTED_ABIS")
+                        .get(null) as Array<String>
+
+                    supportedAbis.firstOrNull()
+                }.getOrNull()
+
+                if (androidAbi != null) {
+                    return "/jni/$androidAbi/$name.so"
+                }
+
+                // Fallback to the default path
+                "/linux/$architecture/$name.so"
+            }
+
+            else ->
+                throw UnsupportedPlatform(operatingSystem.name, architecture)
```

Comment:
note: I've checked returned lib path for device with `arm64-v8a` as supported abi (for temporary purpose method wasn't private, hence I could call it):

<img width="1022" alt="image" src="https://github.com/user-attachments/assets/a4ea21fc-d1e9-4f43-b80a-495091840a38" />